packages/merkle-distributor/scripts/1_CreateClaimsForWindow.ts

```diff
@@ -0,0 +1,84 @@
+//     "metaData": { "reason": ["YD-WETH-21 Liquidity Mining Week 27"] }
+//   }... for all recipients
+// }
+
+// example execution: ts-node ./scripts/1_CreateClaimsForWindow.ts -i ./scripts/example.json
+
+const assert = require("assert");
+const path = require("path");
+import { program } from "commander";
+import fs from "fs";
+const { getWeb3 } = require("@uma/common");
+const web3 = getWeb3();
+const { toBN } = web3.utils;
+
+import MerkleDistributorHelper = require("../src/MerkleDistributorHelper");
+
+program
+  .requiredOption("-i, --input <path>", "input JSON file location containing a recipients payout")
+  .parse(process.argv);
+
+const options = program.opts();
+
+const recipientsObject = JSON.parse(fs.readFileSync(options.input, { encoding: "utf8" }));
+
+// We can't easily do runtime verification of JSON file types in typescript. We could use a JSON scheme, but to keep
+// things simple for now we can just double check that some important keys are present within the JSON file.
+if (typeof recipientsObject !== "object") throw new Error("Invalid JSON");
+const expectedKeys = ["chainId", "rewardToken", "windowIndex", "totalRewardsDistributed", "windowStart", "recipients"];
+expectedKeys.forEach(expectedKey => {
+  if (!Object.keys(recipientsObject).includes(expectedKey)) {
+    throw new Error(`recipients object missing expected key: ${expectedKey}`);
+  }
+});
+
+async function main() {
+  console.log("Running claims creation script 🧞‍♂️");
+
+  // Do some basic sanity checks. In particular, verify that totalRewardsDistributed equals the sum of all amounts.
+  let totalCalculatedRewards = toBN("0");
+  Object.keys(recipientsObject.recipients).forEach((recipientAddress: any) => {
+    totalCalculatedRewards = totalCalculatedRewards.add(toBN(recipientsObject.recipients[recipientAddress].amount));
+  });
+  assert(totalCalculatedRewards.toString() == recipientsObject.totalRewardsDistributed, "Wrong total rewards");
+
+  // Generate the merkle proofs for each recipient & the merkle root of the tree.
+  const { recipientsDataWithProof, merkleRoot } = MerkleDistributorHelper.createMerkleDistributionProofs(
+    recipientsObject.recipients,
+    recipientsObject.windowIndex
+  );
+
+  // Append the merkleRoot & claims recipientsDataWithProof to a new output format
+  const outputData: any = {
+    chainId: recipientsObject.chainId,
+    rewardToken: recipientsObject.rewardToken,
+    windowIndex: recipientsObject.windowIndex,
+    totalRewardsDistributed: recipientsObject.totalRewardsDistributed,
+    windowStart: recipientsObject.windowStart,
+    merkleRoot,
+    claims: recipientsDataWithProof
+  };
+
+  // Save the file to disk.
+  const savePath = `${path.resolve(__dirname)}/../proof-files/chain-id-${recipientsObject.chainId}-reward-window-${
+    recipientsObject.windowIndex
+  }-claims-file.json`;
+  fs.writeFileSync(savePath, JSON.stringify(outputData));
+  console.log("🗄  File successfully written to", savePath);
+}
+
+main();
```

Comment:
my testing it plays just fine. the only _small_ issue is that it does not exit if it crashes and you need to `ctrl+c` to end the process. I played around with a `.then` & `.catch` pattern but it did not always handle error cases correctly always :/ it could be a type script artefact, I'm not sure.

I left it like this because it is min code to achieve the goal while still being 100% compatible with the console.

Reply:
I've refactored this to use:
```
main().catch(e => {
  console.log(e);
  process.exit(1);
});

```
and it works great like this.

Reply:
Sounds good. All I wanted to make sure is that:
- The process terminates itself in all cases.
- The exit code is correct (i.e. errors are reported).